src/bin/commands/demux.rs

```diff
@@ -19,42 +19,44 @@ use serde::{Deserialize, Serialize};
 use std::collections::HashSet;
 use std::fs::File;
 use std::io::{BufRead, BufWriter, Write};
+use std::iter::Filter;
+use std::slice::Iter;
 use std::{
     fs,
     path::{Path, PathBuf},
 };
 
-type VecOfReaders = Vec<Box<dyn BufRead>>;
+/// Type alias to prevent clippy complaining about type complexity
+type VecOfReaders = Vec<Box<dyn BufRead + Send>>;
+
+/// Type alias for segment type iter functions, which iterate over the segments of a ``ReadSet``
+/// filtering for a specific type.
+type SegmentTypeIter<'a> = Filter<Iter<'a, FastqSegment>, fn(&&FastqSegment) -> bool>;
```

Comment:
```suggestion
type SegmentIter<'a> = Filter<Iter<'a, FastqSegment>, fn(&&FastqSegment) -> bool>;
```

since it's iterator over segments, not segment types